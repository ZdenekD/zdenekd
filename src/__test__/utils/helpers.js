import React from 'react';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';

/**
 * Factory function to create a shallow wrapper for App component
 * @function setup
 * @param {ReactComponent} Component - React component to render with shallow
 * @param {Object} props - Component props specific to this setup
 * @param {Object} state - Initial state for setup
 * @returns {ShallowWrapper}
 */
export const factory = (Component, props = {}, state = null) => {
    const wrapper = shallow(<Component {...props} />);

    if (state) {
        wrapper.setState(state);
    }

    return wrapper;
};

/**
 * Return ShallowWrapper containg node(s) with the data-test value
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within
 * @param {string} value - data attribute value to find
 * @returns {ShallowWrapper}
 */
export const findComponent = (wrapper, value) => wrapper.find(`[data-test="${value}"]`);

/**
 * Check corret properties in component
 * @param {ReactComponent} component
 * @param {object} props
 * @returns {Error|undefined}
 */
export const checkProps = (component, props) => checkPropTypes(component.propTypes, props, 'props', component.name);
