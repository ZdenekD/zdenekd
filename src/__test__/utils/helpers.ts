import {ShallowWrapper} from 'enzyme';

const findComponent = (wrapper: ShallowWrapper, attribute: string): ShallowWrapper => wrapper.find(`[data-test="${attribute}"]`);

export default findComponent;
