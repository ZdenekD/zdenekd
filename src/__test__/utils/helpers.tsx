import {
    mount,
    ReactWrapper,
    ShallowWrapper
} from 'enzyme';
import {Provider} from '../../state';

type IWrapper = ShallowWrapper | ReactWrapper;

const findComponent = (wrapper: IWrapper, attribute: string): IWrapper => wrapper.find(`[data-test="${attribute}"]`);

export const stateFactory = (children: React.ReactNode): IWrapper => {
    const wrapper = mount(
        <Provider>
            {children}
        </Provider>
    );

    return wrapper;
};

export default findComponent;
