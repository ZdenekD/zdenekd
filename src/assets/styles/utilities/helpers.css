:global ._visuallyhidden {
    width: 1px;
    height: 1px;
    margin: -1px;
    padding: 0;
    position: absolute;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    border: none;
}

:global ._align-right {
    text-align: right !important;
}
