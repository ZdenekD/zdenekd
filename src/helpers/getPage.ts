import pages from '../data/pages';

const getPage = (link: string): string => Object.keys(pages).filter(item => `/${pages[item].slug}` === link)[0];

const getPageIndex = (link: string): number => Object.keys(pages).findIndex(item => `/${pages[item].slug}` === link);

export {
    getPage,
    getPageIndex
};
