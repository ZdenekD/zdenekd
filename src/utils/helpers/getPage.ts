import pages from '@/data/pages';
import PagesEnum from '@/enums/PagesEnum';

const getPage = (link: string): string => Object.keys(pages).filter(item => pages[item as PagesEnum].route === link)[0];

export default getPage;
