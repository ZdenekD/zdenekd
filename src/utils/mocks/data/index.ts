import MockDataParagraph from './content/UI/paragraph';

export {MockDataParagraph};
