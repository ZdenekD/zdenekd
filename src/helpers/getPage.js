import data from '../data/pages';

/**
 * Get page data object to work with in component
 * @param {string} link - URL link value to find in data object
 * @returns {string}
 */
const getPage = link => Object.keys(data).filter(item => data[item].link === link)[0];

/**
 * Get page index from data object
 * @param {string} link - URL link value to find index
 * @returns {number}
 */
const getPageIndex = link => Object.keys(data).findIndex(item => data[item].link === link);

export {
    getPage,
    getPageIndex
};
