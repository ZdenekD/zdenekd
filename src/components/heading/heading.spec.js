import React from 'react';
import {mount} from 'enzyme';
import {findComponent, getStore} from '../../__test__/utils/helpers';
import Heading from './index';
import State from '../../state';
import data from '../../data/pages';
import {getPage} from '../../helpers/getPage';

jest.mock('next/router', () => ({useRouter: jest.fn().mockImplementation(() => ({route: '/'}))}));

describe('Heading', () => {
    const store = getStore({menu: {isOpen: false, isAnimated: false}});
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <State.StateProvider>
                <Heading store={store} />
            </State.StateProvider>
        );
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-heading');

        expect(component.exists()).toBe(true);
    });

    it('renders appropriate heading title', () => {
        const component = findComponent(wrapper, 'component-title');
        const page = getPage('/');
        const heading = data[page].title;

        expect(component.text().includes(heading)).toBe(true);
    });
});
