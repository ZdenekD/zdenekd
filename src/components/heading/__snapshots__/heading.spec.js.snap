// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading match snapshot 1`] = `
<a href="/"
   aria-label="Odkaz na domovskou stránku"
>
  <h1 data-test="component-heading">
    <span class="_visuallyhidden"
          data-test="component-title"
    >
      Úvodní stránka | ZdenekD
    </span>
    <svg data-filename="src/assets/images/logo.svg">
    </svg>
    <i class="undefined ">
    </i>
  </h1>
</a>
`;
